cmake/define-test.cmake

```diff
@@ -54,37 +54,22 @@ function(define_shared_lib)
     set(COPIED_SRCS ${SHARED_LIB_SRCS})
     list(TRANSFORM COPIED_SRCS PREPEND ${CMAKE_CURRENT_BINARY_DIR}/)
 
-    if(DEFINED SHARED_LIB_INCLUDE_DIR)
-        file(GLOB ORIGINAL_LIB_HEADERS "${ORIGINAL_INCLUDE_DIR}/*.h")
-        list(TRANSFORM ORIGINAL_LIB_HEADERS REPLACE
-            ${ORIGINAL_INCLUDE_DIR} ${REWRITTEN_INCLUDE_DIR}
-            OUTPUT_VARIABLE COPIED_LIB_HEADERS)
-        add_custom_command(
-            OUTPUT ${COPIED_SRCS} ${COPIED_LIB_HEADERS}
-            COMMAND cp ${ORIGINAL_SRCS} ${CMAKE_CURRENT_BINARY_DIR}/
-            COMMAND mkdir -p ${REWRITTEN_INCLUDE_DIR}
-            COMMAND cp ${ORIGINAL_LIB_HEADERS} ${REWRITTEN_INCLUDE_DIR}/
-            DEPENDS ${ORIGINAL_SRCS} ${ORIGINAL_LIB_HEADERS}
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-        add_custom_target(copy_files_${LIBNAME} DEPENDS ${COPIED_SRCS} ${COPIED_LIB_HEADERS})
-    else()
-        add_custom_command(
-            OUTPUT ${COPIED_SRCS}
-            COMMAND cp ${ORIGINAL_SRCS} ${CMAKE_CURRENT_BINARY_DIR}/
-            DEPENDS ${ORIGINAL_SRCS}
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-        add_custom_target(copy_files_${LIBNAME} DEPENDS ${COPIED_SRCS})
-    endif()
-
     # Define two targets
     set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-    add_library(${LIBNAME} SHARED ${COPIED_SRCS})
+    add_library(${LIBNAME} SHARED ${ORIGINAL_SRCS})
     set(CMAKE_EXPORT_COMPILE_COMMANDS OFF)
     add_library(${WRAPPED_LIBNAME} SHARED ${COPIED_SRCS})
     add_dependencies(${WRAPPED_LIBNAME} ${TEST_NAME}_call_gate_generation)
     set_target_properties(${LIBNAME} PROPERTIES EXCLUDE_FROM_ALL 1)
 
     target_compile_definitions(${LIBNAME} PRIVATE PRE_REWRITER=1)
+    if (${SHARED_LIB_PKEY} GREATER 0)
+        target_include_directories(${LIBNAME} BEFORE PRIVATE ${ORIGINAL_INCLUDE_DIR})
+        target_include_directories(${WRAPPED_LIBNAME} BEFORE PRIVATE ${REWRITTEN_INCLUDE_DIR})
```

Comment:
Shouldn't this be under the foreach?

Reply:
No, we only want to use the rewritten headers for the wrapped target.